python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py

```diff
@@ -306,6 +330,31 @@ def _should_forward_logs(self) -> bool:
         # Airbyte Cloud does not support streaming logs yet
         return False
 
+    def _refresh_access_token(self) -> None:
+        response = check.not_none(
+            self.make_request(
+                endpoint="/applications/token",
+                data={
+                    "client_id": self.client_id,
+                    "client_secret": self.client_secret,
+                },
+                # Must not pass the bearer access token when refreshing it.
+                include_additional_request_params=False,
+            )
+        )
+        self._access_token_value = str(response["access_token"])
+        self._access_token_timestamp = datetime.now().timestamp()
+
+    def _needs_refreshed_access_token(self) -> bool:
```

Comment:
This behavior makes sense to me, this is low enough complexity that it feels ok to abstract from the user